editor/components/EMSLinkRecordButton.tsx

```diff
@@ -0,0 +1,53 @@
+import Button from "react-bootstrap/Button";
+import { EMSPatientCareRecord } from "@/types/ems";
+import { RowActionComponentProps } from "@/components/RelatedRecordTable";
+import { FaLink } from "react-icons/fa6";
+import PermissionsRequired from "@/components/PermissionsRequired";
+import AlignedLabel from "@/components/AlignedLabel";
+
+const allowedLinkRecordRoles = ["vz-admin", "editor"];
+
+export interface EMSLinkRecordButtonProps extends Record<string, unknown> {
+  onClick: (emsId: EMSPatientCareRecord) => void;
+  selectedEmsPcr: EMSPatientCareRecord | null;
+}
```

Comment:
Here's how we can define the type signature of the abstract `RowActionComponent`. This typing is handed to the `RelatedRecordTable` as a generic that makes its way down to the tree to the `RelatedRecordTableRow`